[ci skip] update ActiveStorage documentation

zealot128 opened the pull request:
### Summary

I've added some parts of documentation that took me some time to figure out by myself. 

Not sure about all the language, as I am not a native speaker.

zealot128 committed:
[ci skip] update ActiveStorage documentation

- added documentation on how to download files, with example of
  ActiveStorage::Downloading
- documentation about linking files outside of controller/view
- added section about DirectUpload JavaScript integration into
  libraries/frameworks, as well as usage in combination with Drag and Drop

georgeclaghorn reviewed (changes requested):
Thanks, @zealot128! These are good additions.

zealot128 commented:
@georgeclaghorn Thank you for your feedback. I've integrated all your recommendations.

While working on some ActiveStorage solutions at work, I've also stumbled upon an issue and Stackoverflow comment from you on how to use the JavaScript DirectUpload class. With that I was able to integrate AS into an Vue.js uploader with Drag&Drop.
I think that snippets you wrote are also very good candidates for the documenation, thus I've added another section in the JS. I also expanded upon the example and added a little fluff which should make it easier for other Googler's :)

The layout of the TOC now looks like this:

![new doc menu](https://user-images.githubusercontent.com/147175/39332121-ce47ba78-49a6-11e8-932e-3aa25b8e8c8e.png)

Looking forward for further feedback!

georgeclaghorn reviewed (commented):
Here are a few more revisions.

georgeclaghorn commented on guides/source/active_storage_overview.md:
````diff
@@ -299,6 +299,16 @@ Call `images.attached?` to determine whether a particular message has any images
+unit-tests, you have to pass the file as keyword arguments:
+
+```ruby
+@message.image.attach(io: File.open('/path/to/file'), filename: 'file.pdf')
````
To be clear, I think you should append that text to the section, like this:

> Sometimes you need to attach a file that doesn’t arrive via an HTTP request. For example, you may want to attach a file you generated on disk or downloaded from a user-submitted URL. You may also want to attach a fixture file in a model test. To do that, provide a Hash containing at least an open IO object and a filename:
>
> ```ruby
> @message.image.attach(io: File.open('/path/to/file'), filename: 'file.pdf')
> ```
>
> When possible, provide a content type as well. Active Storage attempts to determine a file’s content type from its data. It falls back to the content type you provide if it can’t do that.
>
> ```ruby
> @message.image.attach(io: File.open('/path/to/file'), filename: 'file.pdf', content_type: 'application/pdf')
> ```
>
> If you don’t provide a content type and Active Storage can’t determine the file’s content type automatically, it defaults to `application/octet-stream`.

zealot128 commented:
OK, once again, I've pasted your comment verbatim 😅 

![screenshot](https://user-images.githubusercontent.com/147175/39357238-8b4cd890-4a12-11e8-99cd-fedb6cac353f.png)

georgeclaghorn commented:
Can you please squash your commits into one?

zealot128 commented:
ok, done it!
